### PR TITLE
community[neo4j_vector]: make embedding dimension check optional

### PR DESCRIPTION
**Pull request by HollowMan6**
**Description:**

Starting from Neo4j 5.23 (22 August 2024), with vector-2.0 indexes, `vector.dimensions` is not required to be set, which will cause it the key not exist error in index config if it's not set.

Since the existence of vector.dimensions will only ensure additional checks, this commit turns embedding dimension check optional, and only do checks when it exists (not None).

https://neo4j.com/release-notes/database/neo4j-5/

**Twitter handle:** @HollowM186

### CODE REVIEW

**Review by baskaryan — COMMENTED** [review]
cc @tomasonjo

**Comment by tomasonjo** [comment]
Thanks for raising this. I need a couple of days to think how we want to
handle this and then I'll get back to you

V sre., 28. avg. 2024, 16:47 je oseba Bagatur ***@***.***>
napisala:

File: libs/community/langchain_community/vectorstores/neo4j_vector.py [hunk 1]
```diff
@@ -1030,9 +1033,12 @@ def similarity_search_with_score_by_vector(
             base_index_query = parallel_query + (
                 f"MATCH (n:`{self.node_label}`) WHERE "
                 f"n.`{self.embedding_node_property}` IS NOT NULL AND "
-                f"size(n.`{self.embedding_node_property}`) = "
```
Comment (tomasonjo) [inline libs/community/langchain_community/vectorstores/neo4j_vector.py:1033]: The embedding dimension comes from embedding model and not index definition, so this is not needed:

https://github.com/langchain-ai/langchain/blob/2ddf5b27f0e6f3c4d4a670f28a0e6f0860a6c575/libs/community/langchain_community/vectorstores/neo4j_vector.py#L576

**Review by tomasonjo — APPROVED** [review]

**Comment by tomasonjo** [comment]
@HollowMan6 thanks for you contribution, we can merge it in as is, and then make any changes later if needed

File: libs/community/langchain_community/vectorstores/neo4j_vector.py (hunk 1 above)
Comment (HollowMan6) [reply libs/community/langchain_community/vectorstores/neo4j_vector.py]: Removed, thank you!

**Comment by efriis** [comment]
last minute edits look ok to me - @tomasonjo if you want to take a look as well

**Comment by tomasonjo** [comment]
Lgtm

V sob., 31. avg. 2024, 06:04 je oseba Erick Friis ***@***.***>
napisala:

**Review by ccurme — APPROVED** [review]

**Pull request merged**